service/worker/scheduler/workflow.go

```diff
@@ -473,8 +479,24 @@ func (s *scheduler) sleep(nextWakeup time.Time) {
 	refreshCh := workflow.GetSignalChannel(s.ctx, SignalNameRefresh)
 	sel.AddReceive(refreshCh, s.handleRefreshSignal)
 
-	// if we're paused or out of actions, we don't need to wake up until we get an update
-	if s.tweakables.SleepWhilePaused && !s.canTakeScheduledAction(false, false) {
+	shouldSelfDestruct := false
+	selfDestruct := func() {
+		if shouldSelfDestruct {
+			ctx := workflow.WithLocalActivityOptions(s.ctx, defaultLocalActivityOptions)
+			err := workflow.ExecuteLocalActivity(ctx, s.a.DeleteScheduleWorkflow, s.State.ScheduleId).Get(s.ctx, nil)
+			if err != nil {
+				//TODO: do we want to retry here?
+				s.logger.Error("Failed to delete schedule workflow", "err", err)
+			}
+		}
+	}
+	// if run out of actions or no more jobs to run and the schedule workflow is not paused, start a timer to delete the schedule workflow
+	if s.hasMinVersion(DeleteIdleSchedule) && (!s.Schedule.State.Paused && (nextWakeup.IsZero() || !s.canTakeScheduledAction(false, false))) {
+		shouldSelfDestruct = true
+		nextWakeup = time.Now().Add(s.tweakables.RetentionTime)
+	}
+	// if we're paused, we don't need to wake up until we get an update
+	if s.tweakables.SleepWhilePaused && s.Schedule.State.Paused {
```

Comment:
I think this logic change breaks determinism? previously if we came in here idle but not paused, we would not set a timer, but now we do?

based on the comment above I'm thinking more like:
```
  func (s *scheduler) sleep(nextWakeup time.Time) {
    ...
    // leave this the same as before:
    if s.tweakables.SleepWhilePaused && !s.canTakeScheduledAction(false, false) { nextWakeup = time.Time{} }
    if s.hasMinVersion(DeleteIdleSchedule) && !s.Schedule.State.Paused && nextWakeup.IsZero()  {
      nextWakeup = s.lastWorkflowStarted + s.tweakables.RetentionTime
    }
    ...
  }
```
to me that's at least easier to see that old versions use unchanged logic